### PR TITLE
Ports Bodycameras from Fulp

### PR DESCRIPTION
**Pull request by Zevotech**
## About The Pull Request
Ports a player-attachable camera that can be viewed on security camera consoles from fulpstation with some adjustments and fixes to make it work on our codebase. 
Also ports most of https://github.com/tgstation/tgstation/pull/52767 to fix movable cameras being a black screen.

The body camera is attached to a jumpsuit, and enabled using an ID. The jumpsuit must be worn for the camera to be active. The body camera can be used on a security camera console to connect it to the network of the console.

They can be found on ships that have security camera consoles or in the security section of cargo for 2000 credits per box.

TODO:

~~Make modcomps connectable to security camera console networks and maybe start with the security camera network of the area they spawn in~~
- [x] Possibly get better sprites for the body cameras (Sort of half done. May get one from a better spriter)
- [x] Add body cameras to maps that would have them
Maps with body cameras: Mudskipper, Riggs, Shetland, Talos, Osprey, Ranger, Crying Sun, Chronicle, Li Tieguai, Komodo, Hyena

## Why It's Good For The Game
You can now monitor individuals from afar using a security camera console! Good for mining vessels and military vessels, as well as any vessel with a dedicated, well-stocked security force. Great for coordination!

Potential changes: 
1. May try to find away to make them armor-mounted without the issues that come with them being stored in the armor inventory.
2. Due to modcomps not connecting to ship camera networks, they don't work with SecureEye. May look into a potential fix but otherwise I want to avoid touching modcomp code.
3. Needs better sprites, possibly an icon for the jumpsuit to indicate it has a bodycam attached.

## Changelog

:cl:
add: adds Bodycameras, ported from fulpstation
add: adds body cameras to the Mudskipper, Riggs, Shetland, Talos, Osprey, Ranger, Crying Sun, Chronicle, Li Tieguai, Komodo, and Hyena
fix: Ports a fix from tg that fixed movable cameras being a black screen
/:cl:

### CODE REVIEW

**Comment by Sun-Soaked** [comment]
aegis security turn off bodycam macro

File: code/modules/clothing/body_camera.dm [hunk 1]
```diff
@@ -0,0 +1,145 @@
+/obj/item/clothing/under/Initialize(mapload)
+	. = ..()
+	AddComponent(/datum/component/bodycamera_holder)
```
Comment (FalloutFalcon) [inline code/modules/clothing/body_camera.dm:3]: I dont like this being here, it should be in the real Init.

File: code/game/machinery/computer/camera.dm [hunk 2]
```diff
@@ -193,6 +197,11 @@
 			D["[C.c_tag]"] = C
 	return D
 
+/obj/machinery/computer/security/attackby(obj/item/bodycam/bc, mob/user, params)
+		bc.cameranetwork = network
+		user.balloon_alert(user, "body camera linked to network.")
+		return
+
```
Comment (FalloutFalcon) [inline code/game/machinery/computer/camera.dm]: This runtimes if you attack it with something that is not a body-cam and prevents you from attacking it
```suggestion
/obj/machinery/computer/security/attackby(obj/item/I, mob/user, params)
	if(istype(I, obj/item/bodycam))
		var/obj/item/bodycam/bodycam = I
		bodycam.cameranetwork = network
		user.balloon_alert(user, "body camera linked to network.")
	else
		return ..()
```

**Review by FalloutFalcon — CHANGES_REQUESTED** [review]

**Comment by rye-rice** [comment]
sowwy, nwo prwogwess iwn mwonths, rweopwen wwen wrorking on iwt agwian

**Pull request closed without merging**